motto/thread_agents/thread_agents.py

```diff
@@ -1,44 +1,148 @@
+import time
 from hyperon import *
-from motto.agents import DialogAgent
+from motto.agents import DialogAgent, MettaAgent
 from hyperon.ext import register_atoms
+from hyperon.atoms import GroundedAtom
+from queue import Queue
+import threading
+from motto.utils import *
 
 class AgentArgs:
     def __init__(self, message, functions=[], additional_info=None):
-        self.message = message
-        self.additional_info = additional_info
-        self.functions = functions
+        self.message = message.get_object().content if isinstance(message, GroundedAtom) else message
```

Comment:
a minor thing, but `x.get_object().content if isinstance(x, GroundedAtom) else x` could be put somewhere as a function